front/lib/webcrawler.ts

```diff
@@ -32,11 +32,11 @@ export function urlToDataSourceName(url: string) {
 
 const urlValidationRegex = new RegExp(
   "^(https?:\\/\\/)?" + // validate protocol
-    "((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|" + // validate domain name
+    "((([a-z\\d@]([a-z\\d@-]*[a-z\\d@])*)\\.)+[a-z]{2,}|" + // validate domain name with @ allowed
     "((\\d{1,3}\\.){3}\\d{1,3}))" + // validate OR ip (v4) address
-    "(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*" + // validate port and path
-    "(\\?[;&a-z\\d%_.~+=-]*)?" + // validate query string
-    "(\\#[-a-z\\d_]*)?$",
+    "(\\:\\d+)?(\\/[-a-z\\d%@_.~+]*)*" + // validate port and path
```

Comment:
i'm not 100% sure here.

Reply:
@Fraggle should we just use `https://github.com/dust-tt/dust/blob/52af1a6e87172d0367e4b9ab6dcd50f721afd47d/types/src/shared/utils/url_utils.ts`?

Reply:
why not :)

Reply:
@Fraggle this should at least be more homogeneous 😄

Reply:
and we love homogeneous things :)